frame/staking/src/pallet/mod.rs

```diff
@@ -559,6 +646,179 @@ pub mod pallet {
 	#[pallet::getter(fn current_planned_session)]
 	pub type CurrentPlannedSession<T> = StorageValue<_, SessionIndex, ValueQuery>;
 
+	/// Wrapper struct for Era related information. It is not a pure encapsulation as these storage
+	/// items can be accessed directly but nevertheless, its recommended to use `EraInfo` where we
+	/// can and add more functions to it as needed.
+	pub(crate) struct EraInfo<T>(sp_std::marker::PhantomData<T>);
+	impl<T: Config> EraInfo<T> {
+		/// Temporary function which looks at both (1) passed param `T::StakingLedger` for legacy
+		/// non-paged rewards, and (2) `T::ClaimedRewards` for paged rewards. This function can be
+		/// removed once `$HistoryDepth` eras have passed and none of the older non-paged rewards
+		/// are relevant/claimable.
+		// Refer tracker issue for cleanup: #13034
+		pub(crate) fn is_rewards_claimed_temp(
+			era: EraIndex,
+			ledger: &StakingLedger<T>,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> bool {
+			ledger.legacy_claimed_rewards.binary_search(&era).is_ok() ||
+				Self::is_rewards_claimed(era, validator, page)
+		}
+
+		/// Check if the rewards for the given era and page index have been claimed.
+		///
+		/// This is only used for paged rewards. Once older non-paged rewards are no longer
+		/// relevant, `is_rewards_claimed_temp` can be removed and this function can be made public.
+		fn is_rewards_claimed(era: EraIndex, validator: &T::AccountId, page: PageIndex) -> bool {
+			ClaimedRewards::<T>::get(era, validator).binary_search(&page).is_ok()
+		}
+
+		/// Get exposure info for a validator at a given era and page.
+		///
+		/// This builds a paged exposure from `ExposureOverview` and `ExposurePage` of the
+		/// validator. For older non-paged exposure, it returns the clipped exposure directly.
+		pub(crate) fn get_validator_exposure(
+			era: EraIndex,
+			validator: &T::AccountId,
+			page: PageIndex,
+		) -> Option<ExposureExt<T::AccountId, BalanceOf<T>>> {
+			match <ErasStakersPaged<T>>::get(era, (validator, page)) {
+				// return clipped exposure if page zero and paged exposure does not exist
+				None if page == 0 =>
+					Some(ExposureExt::from_clipped(<ErasStakersClipped<T>>::get(era, validator))),
+
+				// return paged exposure if it exists
+				Some(exposure_page) => {
+					let overview = <ErasStakersOverview<T>>::get(&era, validator);
+					// own stake is included only once in the first page.
+					let own = if page == 0 { overview.own } else { Zero::zero() };
+
+					Some(ExposureExt {
+						exposure_overview: ExposureOverview { own, ..overview },
+						exposure_page,
+					})
+				},
+				_ => None,
+			}
+		}
+
+		/// Returns the number of pages of exposure a validator has for the given era.
+		///
+		/// This will always return at minimum one count of exposure to be backward compatible to
+		/// non-paged reward payouts.
```

Comment:
Maybe it's worth to add a comment here to remove the `page_count.max(1)` after removing the old logic. This comes across as a piece of code that will easily be missed when refactoring.